### PR TITLE
remove $self = $this; use ($self) in 5.4

### PR DESCRIPTION
**Pull request by samsonasik**
ping @carnage I created a PR for your suggestion.

### CODE REVIEW

File: library/Zend/ServiceManager/AbstractPluginManager.php [hunk 1]
```diff
@@ -62,10 +62,9 @@
     public function __construct(ConfigInterface $configuration = null)
     {
         parent::__construct($configuration);
-        $self = $this;
-        $this->addInitializer(function ($instance) use ($self) {
+        $this->addInitializer(function ($instance) {
             if ($instance instanceof ServiceLocatorAwareInterface) {
-                $instance->setServiceLocator($self);
+                $instance->setServiceLocator($this);
```
Comment (samsonasik) [inline library/Zend/ServiceManager/AbstractPluginManager.php]: it makes test error, 
```
1) ZendTest\ServiceManager\AbstractPluginManagerTest::testValidatePluginIsCalledWithDelegatorFactoryIfItsAService

Undefined variable: serviceManager
```
maybe rollback it ?

Comment (samsonasik) [inline library/Zend/ServiceManager/AbstractPluginManager.php]: oh, its strange, seems not related with it...

File: library/Zend/ServiceManager/ServiceManager.php [hunk 2]
```diff
@@ -1144,10 +1142,9 @@ protected function checkNestedContextStop($force = false)
      */
     protected function createDelegatorFromFactory($canonicalName, $requestedName)
     {
-        $serviceManager     = $this;
         $delegatorsCount    = count($this->delegators[$canonicalName]);
-        $creationCallback   = function () use ($serviceManager, $requestedName, $canonicalName) {
-            return $serviceManager->doCreate($requestedName, $canonicalName);
+        $creationCallback   = function () use ($requestedName, $canonicalName) {
+            return $this->doCreate($requestedName, $canonicalName);
```
Comment (samsonasik) [inline library/Zend/ServiceManager/ServiceManager.php]: it seems it cause the error ..., i will try to rollback it.

File: library/Zend/ServiceManager/ServiceManager.php [hunk 3]
```diff
@@ -605,12 +605,10 @@ public function create($name)
      */
     private function createDelegatorCallback($delegatorFactory, $rName, $cName, $creationCallback)
     {
-        $serviceManager  = $this;
```
Comment (Ocramius) [inline library/Zend/ServiceManager/ServiceManager.php]: Careful with the delegators: they're messy ;-)

Comment (samsonasik) [inline library/Zend/ServiceManager/ServiceManager.php]: I made a rollback change on https://github.com/samsonasik/zf2/commit/68a626a6bc7dd28356340af4d004a2a70ef686ca , and in my local the test is pass now.

**Comment by samsonasik** [comment]
@Ocramius yay, travis is green now ;)

**Comment by Ocramius** [comment]
@samsonasik merging, thanks!
